### ISSUE TITLE
qwik-speak does not resolve locale 

### ISSUE THREAD

**Issue by primozs** [robisim74/qwik-speak#26]
"@builder.io/qwik": "^0.20.1",
"@builder.io/qwik-city": "^0.5.2",
"qwik-speak": "^0.6.2",

I am setting locale onRequest and in renderToStream
q:locale is not set in html tag. Is this qwik-city bug?

![Screenshot from 2023-03-01 21-30-11](https://user-images.githubusercontent.com/1006789/222258453-07c96c42-597e-426e-9180-976f9c5ae945.png)

**Comment by robisim74** [comment]
Hi @primozs,

this library does not set the `q:locale` in html tag. This is done by Qwik City when you call the function `locale(lang)`.

Without 'q:locale', also this library is broken on first loading. Opened a new issue in Qwik repo: https://github.com/BuilderIO/qwik/issues/3228

Thanks for reporting.

**Comment by robisim74** [comment]
As a temporary workaround, you can manually set the `locale` in `entry.ssr.tsx`:

```
export default function (opts: RenderToStreamOptions) {
  return renderToStream(<Root />, {
    manifest,
    ...opts,
    // Use container attributes to set attributes on the html tag.
    containerAttributes: {
      lang: opts.serverData?.locale || config.defaultLocale.lang,
      ...opts.containerAttributes,
    },
    // Workaround for https://github.com/BuilderIO/qwik/issues/3228
    // Manually set Qwik locale
    serverData: {
      ...opts.serverData,
      locale: opts.serverData?.qwikcity?.params.lang || config.defaultLocale.lang
    },
  });
}
```

The example is for _localized router_. It is possible to adapt it to obtaining the locale in other ways (cookies, headers) using 
 `serverData`.

**Comment by primozs** [comment]
Thank you for this info!

**Comment by robisim74** [comment]
Resolved with latest version: Qwik 0.22.0, Qwik City 0.6.2 and Qwik Speak 0.8.0
